constantTimeCompare.js

```diff
@@ -30,18 +30,17 @@ module.exports = function constantTimeCompare(a, b) {
   // Using with{} nixes some V8 optimizations that would otherwise undermine
   // our intentions here.
   with({}) {
-    // Add at least one character so that there's at least one thing to modulo over.
-    a += " ";
-    b += " ";
     var aLen = a.length,
         bLen = b.length,
-        match = aLen === bLen ? 1 : 0,
+        match = 0,
         i = Math.max(aLen, bLen, MAX_KEY_CHARS);
+    // If the lengths don't match, the string doesn't match
+    if (aLen !== bLen) {
+      return false;
+    }
     while (i--) {
-      // We repeat the comparison over the strings with % so that we do not compare
-      // a number to NaN, since that has different timing that comparing two numbers.
-      match &= a.charCodeAt( i % aLen ) === b.charCodeAt( i % bLen ) ? 1 : 0;
+      match |= a.charCodeAt( i ) ^ b.charCodeAt( i );
```

Comment:
— Got it.

As this is not a general-purpose constant-time comparison it may be appropriate to assume that the strings will be of equal length (i.e. it's only going to be comparing strings from the same hash function in the ordinary course) may be one we're prepared to make.

I'm not positive that we can assume equal string lengths, but if we can then I think this patch makes sense.

To make sure it is clear, I think the function name should change or a big comment should be added to indicate that it is only constant-time for strings of equal length and otherwise short-circuits.